registration/registration.go

```diff
@@ -317,12 +387,51 @@ func (r *Registration) newRoundOpts() []newRoundOptionFunc {
 	}
 }
 
+// verifyCert verifies the certificate is signed by a recognized certifier.
+//
+// The hint is optional, however, without it, we only check the default certifier public key,
+// if it is configured. If the default certifier is not configured, we return an error.
+//
+// If the hint is provided we check certificate signature against all matching keys.
+func (r *Registration) verifyCert(
+	certificate *shared.OpaqueCert,
+	certPubKeyHint *shared.CertKeyHint,
+	nodeID []byte,
+) error {
+	// fallback to the default certifier if no hint is provided and the default certifier is configured
+	if certPubKeyHint == nil {
+		if key := r.cfg.Certifier.PubKey.Bytes(); len(key) != 0 {
+			_, err := shared.VerifyCertificate(certificate, key, nodeID)
+			return err
+		}
+		return ErrNoMatchingCertPublicKeys
+	}
+	r.trustedKeysMtx.RLock()
+	matchingKeys, ok := r.trustedCertifierKeys[*certPubKeyHint]
+	r.trustedKeysMtx.RUnlock()
+
+	if !ok {
+		return ErrNoMatchingCertPublicKeys
+	}
+
+	var certErrors error
+	for _, key := range matchingKeys {
+		_, err := shared.VerifyCertificate(certificate, key, nodeID)
+		if err == nil {
+			return nil
+		}
+		certErrors = errors.Join(err)
+	}
+	return fmt.Errorf("%w: errors: %w", ErrInvalidCertificate, certErrors)
```

Comment:
I think this can be simplified to:

```suggestion
	for _, key := range matchingKeys {
		_, err := shared.VerifyCertificate(certificate, key, nodeID)
		if err == nil {
			return nil
		}
	}
	return ErrInvalidCertificate
```

Reason being that a) all `err` joined to `certErrors` will be the same (some error failing to validate the certificate with the given key) and b) `ErrInvalidCertificate` already implies that the provided certificate failed to validate and adding `certErrors` would just repeat the same error again.